apps/backend/src/user/user.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { UserEntity } from './entities/user.entity';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { UserDto, UserPublicDTO } from '@datatlas/shared/models';
+import { EntityRepository } from '@mikro-orm/core';
+import * as util from 'util';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(UserEntity)
+    private readonly userRepository: EntityRepository<UserEntity>
+  ) {}
+
+  /**
+   * Adds a user into the database.
+   * TODO : passwords are stored clearly in database -> hash them !
+   * @param userDto Private user data.
+   * @return number New user_id or 0 if user already exists.
+   */
+  async createUser(userDto: UserDto): Promise<number> {
+    const user = new UserEntity(userDto.username, userDto.password, userDto.role, userDto.active);
+    return await this.isUsernameAlreadyInDatabase(user.username).then((isItIn) => {
+      if (isItIn) {
+        // User already in database.
+        return 0;
+      }
+      return this.userRepository.persistAndFlush(user).then(() => {
+        return this.getUserIDByUserName(user.username);
+      });
```

Comment:
You could the following shorthand version :
```suggestion
      return this.userRepository.persistAndFlush(user).then(() => this.getUserIDByUserName(user.username));
```